script.js

```diff
@@ -11,3 +45,313 @@ function hideColumnContents(className) {
     }
   }
 }
+
+function adjustWidthOfColumns(firstColumnToBeAdjusted, secondColumnToBeAdjusted, headerOfFirstColumnToBeAdjusted, headerOfSecondColumnToBeAdjusted, firstColumnToBeAdjustedWidth, secondColumnToBeAdjustedWidth) {
+  for (const element of firstColumnToBeAdjusted) {
+    element.style.width = firstColumnToBeAdjustedWidth + 'px';
+    
+  }
+
+  headerOfFirstColumnToBeAdjusted.style.width = firstColumnToBeAdjustedWidth + 'px';
+
+  for (const element of secondColumnToBeAdjusted) {
+    element.style.width = secondColumnToBeAdjustedWidth + 'px';
+    
+  }
+
+  headerOfSecondColumnToBeAdjusted.style.width = secondColumnToBeAdjustedWidth + 'px';
+}
+
+document.addEventListener('DOMContentLoaded', function() {
+  const columnlineOne = document.querySelector('#columnline-one');
+  const columnlineTwo = document.querySelector('#columnline-two');
+  const columnlineThree = document.querySelector('#columnline-three');
+  const columnlineFour = document.querySelector('#columnline-four');
+  const columnlineFive = document.querySelector('#columnline-five');
+  var columnOne = document.getElementsByClassName('table__first-column');
+  var columnTwo = document.getElementsByClassName('table__second-column');
+  var columnThree = document.getElementsByClassName('table__third-column');
+  var columnFour = document.getElementsByClassName('table__fourth-column');
+  var headerOne = document.getElementById('table__first-header');
+  var headerTwo = document.getElementById('table__second-header');
+  var headerThree = document.getElementById('table__third-header');
+  var headerFour = document.getElementById('table__fourth-header');
+  var table = document.getElementById('table-id');
+
+  columnlineOne.onmousedown = function(event) {
+    function moveToMouse(windowCoordinatesX) {
+      const newColumnlineLeft = windowCoordinatesX;
+      const newColumnLeftIsInTable = newColumnlineLeft >= table.offsetLeft;
+
+      if (newColumnLeftIsInTable) {
+        columnlineOne.style.left = windowCoordinatesX + 'px';
+      }
+      else {
+        document.removeEventListener('mousemove', onMouseMove);
+      }
+    }
+
+    moveToMouse(event.clientX);
+
+    function onMouseMove(event) {
+      moveToMouse(event.clientX);
+    }
```

Comment:
Move the declaration of this function outside the function. Maybe even to another file for more readability

Reply:
The function has to be inside the function because I cannot add parameters into a function which is to be referred for an eventListener

Reply:
In order to move this function outside the superior function, I have to hand in parameters but unfortunately that doesn't work for function I use for eventListeners, Yes I could declare everything I declared for the superior function but I can't determine the current column line because I cannot hand in a parameter for that.